docs/kb/semgrep-ci/jenkins-shared-library-with-semgrep.md

```diff
@@ -0,0 +1,119 @@
+---
+tags:
+  - Jenkins
+  - Shared Library
+description: Setting up Jenkins shared library with Semgrep scans
+---
+
+# Jenkins shared library with Semgrep scans
+
+## Motivation
+A good practice when programming is to avoid duplicate code. The main issue with code duplication is not the duplication itself, but when you need to modify a piece of this duplicated code, you need to do it two, three or n-times as many repetitions as you have. And even worse, when the duplicated fragment has a bug or a vulnerability, you need to fix it in all the occurrences.
+Luckily, there is an easy solution for this: encapsulate code in only one place: a method, a function, or a library; it doesn’t matter; the key concept here is removing duplications and replacing them with a simple call.
+
+When defining a Jenkins pipeline, you add the logic to build/test your software; for example, test your software with Semgrep to identify security bugs.
+In this article, we will define a [Jenkins Shared Library](https://www.jenkins.io/doc/book/pipeline/shared-libraries/) with the Semgrep snippet so we can call this library in all the projects we have in our organisation.
+
+## Creating a Jenkins pipeline to call Semgrep
+If you follow [Semgrep documentation](https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-jenkins-configuration-snippet), you can create a simple Jenkins pipeline like this:
```

Comment:
Similarly here - I would just link to the doc showing how to create the single config, and skip reproducing the code snippet and explaining the scenario - rather I'd go right to discussing creating the shared snippet.

Reply:
Ok!